sdk/eventhub/event-hubs/src/batchingReceiver.ts

```diff
@@ -71,29 +71,48 @@ export class BatchingReceiver extends EventHubReceiver {
     let timeOver = false;
 
     const receiveEventPromise = () =>
-      new Promise<ReceivedEventData[]>((resolve, reject) => {
+      new Promise<ReceivedEventData[]>(async (resolve, reject) => {
         let onReceiveMessage: OnAmqpEvent;
         let onReceiveError: OnAmqpEvent;
         let onReceiveClose: OnAmqpEvent;
         let onSessionError: OnAmqpEvent;
         let onSessionClose: OnAmqpEvent;
         let waitTimer: any;
         let actionAfterWaitTimeout: Func<void, void>;
-        // Final action to be performed after maxMessageCount is reached or the maxWaitTime is over.
-        const finalAction = (timeOver: boolean) => {
+
+        const rejectOnAbort = () => {
+          const desc: string =
+            `[${this._context.connectionId}] The request operation on the Receiver "${this.name}" with ` +
+            `address "${this.address}" has been cancelled by the user.`;
+          log.error(desc);
+          reject(new AbortError("The receive operation has been cancelled by the user."));
+        };
+
+        // operation has been cancelled, so exit quickly
+        if (abortSignal && abortSignal.aborted) {
+          return rejectOnAbort();
+        }
+
+        const removeListeners = () => {
```

Comment:
Since this does more than just remove listeners, I would rename this to say `cleanUpBeforeReturn` ?
Or remove the setting of `isReceivingMessages` and clearing of timer from this function